x-pack/plugins/uptime/server/rest_api/synthetics_service/api_key.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { UMRestApiRouteFactory } from '../types';
+import { API_URLS } from '../../../common/constants';
+import { SyntheticsForbiddenError } from '../../lib/synthetics_service/get_api_key';
+
+export const getSyntheticsEnablementRoute: UMRestApiRouteFactory = (libs) => ({
+  method: 'GET',
+  path: API_URLS.SYNTHETICS_ENABLEMENT,
+  validate: {},
+  handler: async ({ request, response, server }): Promise<any> => {
+    try {
+      return response.ok({
+        body: await libs.requests.getSyntheticsEnablement({
+          request,
+          server,
+        }),
+      });
+    } catch (e) {
+      server.logger.error(e);
+      throw e;
+    }
+  },
+});
+
+export const disableSyntheticsRoute: UMRestApiRouteFactory = (libs) => ({
+  method: 'DELETE',
+  path: API_URLS.SYNTHETICS_ENABLEMENT,
+  validate: {},
+  handler: async ({ response, request, server, savedObjectsClient }): Promise<any> => {
+    try {
+      await libs.requests.deleteServiceApiKey({
```

Comment:
@shahzad31 When we disable synthetics, should be send a delete request to the synthetics service to immediately stop indexing monitor results? 

The sync task will encounter [this error](https://github.com/elastic/kibana/pull/128223/files#diff-05f832378ee1977230af36dafea749010bcf43a4ca934543c335a43b207ecbe7R191) when the API key is deleted, which will stop sync from processing. However, the synthetics service will continue to index documents for 15 minutes.

Reply:
I think we should call the endpoint to stop monitors.